management/commands/import_scraptf_donations.py

```diff
@@ -59,71 +66,92 @@ def handle(self, *args, **options):
                  failed=failed,
                  total=len(donations))
 
+    # Can't use latest() here because we fetch ScrapTF donations in bulk, no
+    # guarantees made about order of insertion.
     def get_latest_scraptf_donation(self):
         return Donation.objects.filter(domain__exact='SCRAPTF').order_by('-timereceived').first()
 
     @transaction.atomic
     def commit_scraptf_donation(self, donation):
-        """
-        From a ScrapTF donation, create a Donation. If incentives are
-        present, create all DonationBids and BidSuggestions as well.
-        """
+        '''
+        From a ScrapTF donation, create a Donor and Donation. If
+        incentives are present, create all DonationBids and
+        BidSuggestions as well. We try very hard to ensure everything
+        is valid for insertion. If anything doesn't look right, we
+        fail the entire transaction.
+        '''
+
+        # Create the Donor
+        user = donation['user']
+        email = donation.get('email')
+        if not email:
+            raise Exception('Donation is missing an email.')
+        validate_email(email)
+        alias = user.get('name', '')[:DONOR_ALIAS_MAX_LENGTH]
+
+        donor = Donor(email=email,
```

Comment:
Does this create a new `Donor` for every donation, even ones which use an `email` we have seen before?

Reply:
`Donor` is made to be unique on `paypalemail`, because the tracker assumes that most donations will come in through PayPal. It is possible that multiple `Donor`s with the same `email` can exist, and this is indeed the case in production data.

I'm not too worried about this and honestly, there'd be too much effort involved in making ScrapTF donors unique before we need this live.

Reply:
Kk, sounds good to me then. I approve this PR 👍 .